### PR TITLE
feat(ui-react-native): Add federated authentication

### PR DESCRIPTION
**Pull request by ioanabrooks**
#### Description of changes

This change adds the existing federated buttons to sign in and sign up routes by cherrypicking commit bc73289 from already approved PR https://github.com/aws-amplify/amplify-ui/pull/4165 plus:
* fix for unit tests
* themed styling for the federated buttons component
* example app

Follow-ups:

* add e2e 
* update docs

| Light mode  | Dark mode |
| ------------- | ------------- |
| Simulator Screenshot - iPhone 13 - 2023-10-07 at 17 51 31|Simulator Screenshot - iPhone 13 - 2023-10-07 at 17 52 12|

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/react-native/ios/Podfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require_relative '../node_modules/react-native/scripts/react_native_pods'
 require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
 
-platform :ios, '11.0'
+platform :ios, '13.0'
```
Comment (ioanabrooks) [inline examples/react-native/ios/Podfile:4]: Higher deployment target required by `AmplifyRTNCore`

File: examples/react-native/package.json [hunk 2]
```diff
@@ -13,14 +13,14 @@
...
   "dependencies": {
+    "@aws-amplify/react-native": "console-preview",
     "@aws-amplify/ui": "*",
     "@aws-amplify/ui-react-core": "*",
     "@aws-amplify/ui-react-core-notifications": "*",
     "@aws-amplify/ui-react-native": "*",
     "@react-native-async-storage/async-storage": "^1.17.5",
     "@react-native-community/netinfo": "^8.3.0",
     "@xstate/react": "3.0.0",
-    "amazon-cognito-identity-js": "^5.2.11",
```
Comment (ioanabrooks) [inline examples/react-native/package.json:23]: This is now used in the `ui` package, so added dependency there, also was getting duplicate values errors between this package and `@aws-amplify/react-native` when building the app.

File: packages/react-native/src/primitives/Divider/styles.ts [hunk 3]
```diff
@@ -20,7 +20,7 @@ export const getThemedStyles = (theme: StrictTheme): DividerStyles => {
       ...components?.divider?.label,
     },
     line: {
-      backgroundColor: colors.black,
+      backgroundColor: colors.font.primary,
```
Comment (ioanabrooks) [inline packages/react-native/src/primitives/Divider/styles.ts:23]: Fix for divider in dark mode, line color should match label text color.

File: examples/react-native/metro.config.js [hunk 4]
```diff
@@ -116,6 +116,12 @@ config.resolver.blockList = [
 
 // point to the example app deps for React and native modules deps
 config.resolver.extraNodeModules = {
+  '@aws-amplify/react-native': path.resolve(
+    __dirname,
+    'node_modules/@aws-amplify/react-native'
+  ),
+  'base-64': path.resolve(__dirname, 'node_modules/base-64'),
```
Comment (ioanabrooks) [inline examples/react-native/metro.config.js:123]: Reason for having to add this: https://github.com/aws-amplify/amplify-js/blob/console-preview/packages/core/src/utils/globalHelpers/index.native.ts#L5

File: packages/ui/package.json [hunk 5]
```diff
@@ -45,6 +45,7 @@
     "typecheck": "tsc --noEmit"
   },
   "dependencies": {
+    "amazon-cognito-identity-js": "^5.2.11",
```
Comment (ioanabrooks) [inline packages/ui/package.json:48]: TODO: remove this as it will be deprecated

**Review by thaddmt — APPROVED** [review]
lgtm

**Review by esauerbo — APPROVED** [review]
LGTM!

**Comment by ioanabrooks** [comment]
Our next example apps fail to build, pending fix to be released in next `console-preview` tag.

**Pull request merged**